### ISSUE TITLE
The main contributors for Ankidroid-Wear are inactive now and am started experiencing bugs when I updated my Ankidroid version

### ISSUE THREAD

**Issue by joaquintoral** [ankidroid/Anki-Android#5621]
###### Reproduction Steps

1. Review using AnkiWear on my Ticwatch Pro. Been using it for more than a year now
2. While reviewing, my new cards will randomly keep getting replenished with new unseen ones (e.g. my max new cards per review session is 20, after a couple of cards, let's there's 8 left, it will randomly reset back to 20 for some reason with new unseen cards.
3. It doesn't always happen, but it can happen randomly

Here's the link for the repo
https://github.com/wlky/AnkiDroid-Wear

I tried submitting an issue there and even emailing the creator, but I have no luck. I don't really know where to check in his code about this as he wouldn't answer my inquiries except one time talking about being busy and that's understandable. Any insights would really be appreciated. 

###### Expected Result
Be able to review properly using my Ticwatch pro

###### Actual Result
New cards keep being replenished, rendering reviewing endlessly

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [ x ] character to confirm the points below:*

[x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x] I have searched for similar existing issues here and on the user forum

**Comment by mikehardy** [comment]
I'm sorry but without source code we can't really say what's going on.
If there is use of the API it could be a bug in AnkiDroid (specifically it's API implementation) but we'd need to see the source.

If you can link to a repository with AnkiDroid API calls showing the usage pattern, or make a small repo using the API that shows the problem (maybe based on the 'apisample' https://github.com/ankidroid/apisample) then we could move forward

**Comment by Arthur-Milchior** [comment]
Technically, it seems that the source code was given above. It's a github repo. 
In my case, the two main troubles will be 1. that I don't have a wearable device to test this software with. And that I don't know how to tells ankidroid to simulate both a smartphone and a wearable device, connected together. 
@joaquintoral Did it shows you the same new cards ? Or does ankidroid at least register the review you did and consider the cards you review as reviewed ?

**Comment by mikehardy** [comment]
Ah - I was moving too quickly there. Okay, with source it's maybe possible. Might be a fun tech project for someone to revive it.

I think this is how you'd set up a watch emulator (they are available via Android Studio similar to the phone emulators) to connect to a companion app on a phone:
https://developer.android.com/training/wearables/apps/creating#setting-up-a-phone

Looks to me like you might have more luck building this https://github.com/aquach/AnkiDroid-Wear/commits/dev

based on the network graph https://github.com/wlky/AnkiDroid-Wear/network

I have no time to pick up Android Wear development as a hobby though

**Comment by joaquintoral** [comment]
Thank you @mikehardy and @Arthur-Milchior and sorry for the delayed response.

As for this question

> 
@joaquintoral Did it show you the same new cards ? Or does ankidroid at least register the review you did and consider the cards you review as reviewed ?

It shows me new set of new cards each time it resets, those I've already reviewed will remain in the pool as _Again_ cards if I did not get them correct. It still considers the cards I reviewed as _reviewed_. I hope this helps

I will check the resources mentioned by mikehardy to get started. Thank you again!
